internal/app/withdraw.go

```diff
@@ -0,0 +1,78 @@
+package app
+
+import (
+	"encoding/json"
+	"github.com/poggerr/gophermart/internal/authorization"
+	"github.com/poggerr/gophermart/internal/models"
+	"io"
+	"net/http"
+	"strconv"
+)
+
+func (a *App) Withdraw(res http.ResponseWriter, req *http.Request) {
+	c, err := req.Cookie("session_token")
+	if err != nil {
+		a.sugaredLogger.Info(err)
+		res.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	userID := authorization.GetUserID(c.Value)
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		a.sugaredLogger.Info(err)
+		res.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	var withdraw models.Withdraw
+
+	err = json.Unmarshal(body, &withdraw)
+	if err != nil {
+		a.sugaredLogger.Info(err)
+		res.WriteHeader(http.StatusBadRequest)
+		return
+	}
```

Comment:
код сложно читается из-за большого количества проверок и отдельных несвязанных блоков. Можно выносить что-то в отдельные функции и делать более понятные проверки
```suggestion
	withdraw, err := buildWithdraw(req)
	// ...
```